packages/select/src/vaadin-select-base-mixin.js

```diff
@@ -256,7 +256,8 @@ export const SelectBaseMixin = (superClass) =>
         menuElement.addEventListener(
           'click',
           (e) => {
-            const value = e.target.value;
+            const item = e.composedPath().find((el) => el._hasVaadinItemMixin);
+            const value = item && item.value;
```

Comment:
Can the item be missing? If it's missing, I'm afraid dispatchChangePending can be incorrectly set to true.

Reply:
Updated the logic to check that item isn't missing.